hibernate-core/src/main/java/org/hibernate/sql/ast/tree/from/TableGroup.java

```diff
@@ -41,6 +41,8 @@
 	String getSourceAlias();
 
 	List<TableGroupJoin> getTableGroupJoins();
+	
+	boolean hasNonInnerJoins();
```

Comment:
I would prefer a name like `isOptional` or `isOuterJoined`. Also, as far as I can say, the way this is modeled right now is not entirely correct. IMO this should only be `true` if there is a `TableGroupJoin` referring to this `TableGroup` and that  `TableGroupJoin` uses anything but `INNER` or the parent table group is `optional`. This can be determined at `TableGroup` construction time.

Reply:
I'm going to change the method name and to determine its value as you suggest

Reply:
You should be able to determine this in e.g. `ToOneAttributeMapping#createTableGroupJoin`. It is outer joined if `sqlAstJoinType != INNER || lhs.isOuterJoined()`. You do something similar for the `PluralAttributeMapping` as well and other instantiations of `TableGroup`